src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3104,11 +3108,215 @@ public override IOperation VisitObjectCreation(IObjectCreationOperation operatio
+            {
+                handleInitializer(innerInitializer);
+            }
+
+            _currentInitializedInstance = previousInitializedInstance;
+
+            void handleInitializer(IOperation innerInitializer)
+            {
+                switch (innerInitializer.Kind)
+                {
+                    case OperationKind.MemberInitializer:
+                        handleMemberInitializer((IMemberInitializerOperation)innerInitializer);
+                        return;
+
+                    case OperationKind.SimpleAssignment:
+                        AddStatement(handleSimpleAssignment((ISimpleAssignmentOperation)innerInitializer));
+                        return;
+
+                    case OperationKind.CollectionElementInitializer:
+                        // PROTOTYPE(dataflow): support collection initializers
+                        throw new NotImplementedException();
+
+                    case OperationKind.Invalid:
+                        AddStatement(Visit(innerInitializer));
+                        return;
+
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(innerInitializer.Kind);
+                }
+            }
+
+            IOperation handleSimpleAssignment(ISimpleAssignmentOperation assignmentOperation)
+            {
+                // We spill all arguments to the lhs under the assumption that any of these could change after reading.
+                IOperation left = assignmentOperation.Target;
+                IOperation right;
+
+                switch (left.Kind)
+                {
+                    case OperationKind.FieldReference:
+                    case OperationKind.EventReference:
+                    case OperationKind.PropertyReference:
+                        var memberReference = (IMemberReferenceOperation)left;
+                        IPropertyReferenceOperation propertyReference = null;
+                        ImmutableArray<IArgumentOperation> propertyArguments = ImmutableArray<IArgumentOperation>.Empty;
+
+                        if (memberReference.Kind == OperationKind.PropertyReference &&
+                            !(propertyReference = (IPropertyReferenceOperation)memberReference).Arguments.IsEmpty)
+                        {
+                            var propertyArgumentsBuilder = ArrayBuilder<IArgumentOperation>.GetInstance();
+                            foreach (IArgumentOperation arg in propertyReference.Arguments)
+                            {
+                                // We assume all arguments have side effects and spill them. We only avoid capturing literals, and
+                                // recapturing things that have already been captured once.
+                                IOperation visitedInnerArgument = Visit(arg.Value);
+                                if (visitedInnerArgument.Kind != OperationKind.Literal &&
```

Comment:
>visitedInnerArgument.Kind != OperationKind.Literal && [](start = 36, length = 53)

I wouldn't special case literals here. They don't have any special meaning in context of this node, as opposed to string literals in context of IInterpolatedStringTextOperation for example. Also, consider using VisitAndCapture logic, it looks like the code here is pretty much a duplication of what it is doing.